active/ScrollingView/ScrollingView.md

```diff
@@ -0,0 +1,2584 @@
+    }
+}
+
+private void RequestZoomTimer_Tick(object sender, object e)
+{
+    AdjustScrollingViewZoomFactor();
+}
+
+private void AdjustScrollingViewZoomFactor()
+{
+    _scrollingView.ZoomFrom(
+    zoomFactorVelocity: 12*_verticalJoystick.Value,
+    inertiaDecayRate: 0.975);
+}
+```
+
+*ScrollViewer (existing)*
+
+Not supported
+
+
+
+## Customizing the bring-into-view participation
+
+When the framework raises a FrameworkElement.RequestBringIntoView event, `ScrollingView` instances in the parent chain attempt to participate. Just before a `ScrollingView` launches an animation for its participation, it raises its BringingIntoView event so the listener can customize that participation.
+
+###  Adjust target offset for sticky headers
+
+This example shows a list control that supports sticky headers. It listens to the `ScrollingView`'s BringingIntoView event to adjust the target vertical offset to account for the sticky header's presence which the `ScrollingView` is unaware of.
+
+*`ScrollingView` (new)*
+
+```csharp
+private void ScrollingView_BringingIntoView(ScrollingView scrollingView, ScrollingBringingIntoViewEventArgs e)
+{
+    // Adjust the BringIntoViewRequestedEventArgs, e.RequestEventArgs, based on provided target vertical
+    // offset and the sticky header curves.
+    AdjustOffsetForStickyHeader(e.RequestEventArgs, e.TargetVerticalOffset);
+}
+```
+
+*ScrollViewer (existing)*
+
+Not supported
+
+### Detect end of bring-into-view contribution
+
+This example shows how you can know when a bring-into-view contribution starts and ends in order to temporarily turn off user interactions during the contribution.
+
+*`ScrollingView` (new)*
+
+```csharp
+private void ScrollingView_BringingIntoView(ScrollingView scrollingView, ScrollingBringingIntoViewEventArgs e)
+{
+    _bringIntoViewScrollInfo = e.ScrollInfo;
+    scrollingView.IgnoredInputKind = ScrollingInputKinds.All;
+}
+
+private void ScrollingView_ScrollCompleted(ScrollingView scrollingView, ScrollingScrollCompletedEventArgs e)
+{
+    if (_bringIntoViewScrollInfo == e.ScrollInfo)
+    {
+        scrollingView.IgnoredInputKind = ScrollingInputKinds.None;
+    }
+}
+```
+
+*ScrollViewer (existing)*
+
+Not supported
+
+
+## Customize the anchor selection
+
+The anchor element in a `ScrollingView` is used to maintain the content of the view port when the scrolling content changes. For example in a list, the center item's position on screen can be maintained when items earlier in the list are removed.
+
+The `ScrollingView` raises its AnchorRequested event to allow handlers to influence the anchor selection.
+
+### Set an explicit anchor element
+
+This example shows a Grid in a `ScrollingView`, and uses a custom algorithm to select an anchor based on an internal currency element in the Grid.
+
+*`ScrollingView` (new)*
+
+```csharp
+private void ScrollingView_AnchorRequested(ScrollingView scrollingView, ScrollingAnchorRequestedEventArgs e)
+{
+    e.Anchor = SelectAnchorElement();
```

Comment:
Normally, EventArgs types don't have mutable properties. If there's expected output don't we usually use a "Set" method on them, so this would be "e.SetAnchor(SelectAnchorElement())" ?

Reply:
We're copying the WUX existing object model: https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.anchorrequestedeventargs.anchor

Reply:
I think they're particularly common for "Requested" events, which are basically method calls, and so sometimes need to have return or [out] values.

Reply:
Sure, and the 'out' parameters for events is usually a "args.SetResult(r)" method. See https://docs.microsoft.com/en-us/uwp/api/Windows.Media.Core.MediaBindingEventArgs for a public example, or the Patterns > Deferrals > App Code Example internal wiki section. (Which is followed by a counter-example showing an EventArgs with a "args.Items" property the app is expected to fill.)